### PR TITLE
Remove unnecessary catalog warehouse config in setup.sh

### PR DESCRIPTION
**Pull request by flyrain**
# Description

The warehouse name will always be overwritten by individual test cases.

Fixes # (issue)

## Type of change
- [x] Clean up

# How Has This Been Tested?

# Checklist:

Please delete options that are not relevant.

- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] If adding new functionality, I have discussed my implementation with the community using the linked GitHub issue

### CODE REVIEW

**Review by anuragmantri — APPROVED** [review]
LGTM. Thanks @flyrain

File: regtests/setup.sh [hunk 1]
```diff
@@ -105,13 +105,10 @@ spark.jars.packages org.apache.iceberg:iceberg-spark-runtime-3.5_2.12:1.5.2,org.
 spark.hadoop.fs.s3.impl org.apache.hadoop.fs.s3a.S3AFileSystem
 spark.hadoop.fs.AbstractFileSystem.s3.impl org.apache.hadoop.fs.s3a.S3A
 spark.sql.variable.substitute true
-
```
Comment (RussellSpitzer) [inline regtests/setup.sh:108]: Unrelated whitespace change. I think we should probably also keep this a little split up for clarity

Deps

Hadoop

Driver

Sql - Session

Sql - Catalog

**Review by RussellSpitzer — APPROVED** [review]

File: regtests/setup.sh (hunk 1 above)
Comment (RussellSpitzer) [reply regtests/setup.sh:108]: ^ This can be saved for a later cleanup if someone else who is as nit-picky as me cares :)

Comment (flyrain) [reply regtests/setup.sh]: Removed the change.

File: regtests/setup.sh [hunk 2]
```diff
@@ -111,7 +111,6 @@ spark.driver.extraJavaOptions -Dderby.system.home=/tmp/derby
 spark.sql.catalog.polaris=org.apache.iceberg.spark.SparkCatalog
 spark.sql.catalog.polaris.type=rest
 spark.sql.catalog.polaris.uri=http://${POLARIS_HOST:-localhost}:8181/api/catalog
-spark.sql.catalog.polaris.warehouse=snowflake
```
Comment (dennishuo) [inline regtests/setup.sh:114]: The mechanism in here for "automatically" updating a local conf is to update line 95 and line 103 to increment the version number, so that the next time someone runs the regtests they'll automatically pick up the newest set of config values.

However, there's the risk that if someone manually updated the spark-defaults file with customizations then the update will comment out all such customizations.

I guess it's a balance between having a consistent conf for everyone running the tests vs allowing people to have persistent differences.

If people only run via docker then the point is moot though, so maybe the whole auto-updating mechanic in here can be considered obsolete.

**Review by dennishuo — APPROVED** [review]
Optional suggestion - consider incrementing the TESTCONF_V5 to TESTCONF_V6

**Review by RussellSpitzer — APPROVED** [review]

File: regtests/setup.sh (hunk 2 above)
Comment (flyrain) [reply regtests/setup.sh:114]: For non-docker use case, it makes sense to update the version. I can make the follow-up change.
